Src/WindowsServer/WindowsServer.Shared/FirstChanceExceptionStatisticsTelemetryModule.cs

```diff
@@ -0,0 +1,188 @@
+        {
+            if (disposing)
+            {
+                this.unregisterAction(this.CalculateStatistics);
+                this.metricManager.Dispose();
+            }
+        }
+
+        private void CalculateStatistics(object sender, FirstChanceExceptionEventArgs firstChanceExceptionArgs)
+        {
+            if (executionSyncObject == LOCKED)
+            {
+                return;
+            }
+
+            try
+            {
+                executionSyncObject = LOCKED;
+
+                var exception = firstChanceExceptionArgs?.Exception;
+
+                if (exception == null)
+                {
+                    WindowsServerEventSource.Log.FirstChanceExceptionCallbackExeptionIsNull();
+                    return;
+                }
+
+                WindowsServerEventSource.Log.FirstChanceExceptionCallbackCalled();
+
+                var type = exception.GetType().FullName;
+
+                // obtaining the operation name. At this stage we have no intention to send this telemetry item
+                ExceptionTelemetry fakeTelemetry = new ExceptionTelemetry(exception);
+                this.telemetryClient.Initialize(fakeTelemetry);
+
+                var operation = fakeTelemetry.Context.Operation.Name;
+
+                // obtaining failing method name by walking 1 frame up the stack
+                var failingMethod = new StackFrame(1).GetMethod();
+                var method = failingMethod.DeclaringType.FullName + "." + failingMethod.Name;
+
+                this.TrackStatistcis(type, operation, method);
+            }
+            catch (Exception exc)
+            {
+                try
+                {
+                    WindowsServerEventSource.Log.FirstChanceExceptionCallbackException(exc.ToInvariantString());
+                }
+                catch (Exception)
+                {
+                    // this is absolutely critical to not throw out of this method
+                    // Otherwise it will affect the customer application behavior significantly
+                }
+            }
+            finally
+            {
+                executionSyncObject = UNLOCKED;
+            }
+        }
+
+        private void TrackStatistcis(string type, string operation, string method)
+        {
+            var dimensions = new Dictionary<string, string>();
+            dimensions.Add("type", this.GetDimCappedString(type, this.typeValues));
+            dimensions.Add("method", this.GetDimCappedString(method, this.methodValues));
+
+            if (!string.IsNullOrEmpty(operation))
+            {
+                dimensions.Add("operation", this.GetDimCappedString(operation, this.operationValues));
+            }
+
+            var metric = this.metricManager.CreateMetric("Exceptions Thrown", dimensions);
+
+            metric.Track(1);
+        }
+
+        private string GetDimCappedString(string value, ConcurrentBag<string> capValues)
+        {
+            if (capValues.Count > 100)
```

Comment:
We might want to expose this setting unless it's a hard limitation in the processing pipeline.